model/span.go

```diff
@@ -28,6 +28,25 @@ var (
 	ErrValidDurationRequired = errors.New("valid duration required")
 )
 
+// Baggage holds the interface for interacting with baggage context propagation
+// implementations.
+type Baggage interface {
+	// Init returns a fresh Baggage primed for usage in a request lifecycle.
+	Init() Baggage
```

Comment:
I wonder why this method is needed, isn't `New` good enough? Can we expose an example or test where this method is important?